x-pack/plugin/core/src/main/java/org/elasticsearch/datastreams/logsdb/PatternMatcher.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.datastreams.logsdb;
+
+import java.util.regex.Pattern;
+
+public class PatternMatcher {
+    private final Pattern pattern;
+
+    private PatternMatcher(final Pattern pattern) {
+        this.pattern = pattern;
+    }
+
+    public boolean matches(final String value) {
+        return value != null && pattern.matcher(value).matches();
+    }
+
+    public static PatternMatcher forLogs() {
+        return new PatternMatcher(Pattern.compile("logs-[^-]+-[^-]+"));
```

Comment:
This doesn't match `logs-generic-my-namespace`. Whether dashes are allowed in the namespace is somewhat ambiguous. We should assume that this can happen.

```suggestion
        return new PatternMatcher(Pattern.compile("logs-.+-.+"));
```

Reply:
ok...I thought that we required exactly 2 "-"...is this documented somewhere?

Reply:
Anyway, the regex you suggested does not work...I tested it versus "logs-apache-production" I will replace it with a working one.

Reply:
The index template doesn't have that as a requirement:

https://github.com/elastic/elasticsearch/blob/a8d7f79070df0fe58ebe16a4197493c6e9210dae/x-pack/plugin/core/template-resources/src/main/resources/logs%40template.json#L2

See also https://github.com/elastic/elasticsearch/issues/112936#issuecomment-2368076975 for a discussion about the discrepancy in the ECS spec: https://github.com/elastic/elasticsearch/issues/112936#issuecomment-2368076975.

Reply:
This class is not needed and has already been removed from this PR.

 `Regex.simpleMatch(logs-*-*, dataStreamName)`` is now used instead, which does exactly what happens when index templates are resolved.